openedx/openedx-platform
# [SE-4682] Removing invitation-only and non-catalog courses from search results

## Description

Using the following options allows us to enable searching for courses in the LMS:
`COURSES_ARE_BROWSABLE=true`
`ENABLE_COURSE_DISCOVERY=true`

However, there are two types of courses which we would like to exclude from search results:
- invitation-only courses
- Courses which are not supposed to be shown in the catalog (i.e., catalog_visibility="none")

This PR adds the two fields `invitation_only` and `catalog_visibility` to the indexed data for each course.

It also adds two extra settings: `SEARCH_SKIP_INVITATION_ONLY_FILTERING` and `SEARCH_SKIP_SHOW_IN_CATALOG_FILTERING`. These can be configured to specify whether the filtering on these fields should be performed or not.

## Testing Instructions
**Sandbox**: [LMS](https://pr28518.sandbox.opencraft.hosting/)

The sandbox has four courses with the following names, with their "invitation" and "catalog" settings matching their titles:
`Test - Doesn't show in Catalog - Needs Invitation`
`Test - Doesn't show in Catalog - Doesn't need Invitation`
`Test - Shows in Catalog - Needs Invitation`
`Test - Shows in Catalog - Doesn't need invitation`

By using the search term "test" in the course discovery page, you should only see the `Test - Shows in Catalog - Doesn't need invitation` course.

- - -
**Settings**
```yaml
EDXAPP_FEATURES_EXTRA:
  COURSES_ARE_BROWSABLE: true
  ENABLE_COURSE_DISCOVERY: true
  ENABLE_SEPARATE_ARCHIVED_COURSES: true
EDXAPP_LMS_ENV_EXTRA:
  SEARCH_SKIP_INVITATION_ONLY_FILTERING: false
  SEARCH_SKIP_SHOW_IN_CATALOG_FILTERING: false
```

## Reviewers
